packages/zoe/src/contracts/priceAggregatorChainlink.js

```diff
@@ -0,0 +1,743 @@
+    const quoteAmount = AmountMath.make(quote, quoteKit.brand);
+    const quotePayment = await E(quoteKit.mint).mintPayment(quoteAmount);
+    return harden({ quoteAmount, quotePayment });
+  };
+
+  const { notifier } = makeNotifierKit();
+  const zoe = zcf.getZoeService();
+
+  /**
+   * @typedef {Object} OracleRecord
+   * @property {(timestamp: Timestamp) => Promise<void>=} querier
+   * @property {number} lastSample
+   */
+
+  /** @type {Store<Instance, Set<OracleRecord>>} */
+  const instanceToRecords = makeStore('oracleInstance');
+
+  /**
+   * @param {Object} param0
+   * @param {number} [param0.overrideValueOut]
+   * @param {Timestamp} [param0.timestamp]
+   */
+  const makeCreateQuote = ({ overrideValueOut, timestamp } = {}) =>
+    /**
+     * @param {PriceQuery} priceQuery
+     * @returns {ERef<PriceQuote>=}
+     */
+    function createQuote(priceQuery) {
+      // Sniff the current baseValueOut.
+      const valueOutForUnitIn =
+        overrideValueOut === undefined
+          ? lastValueOutForUnitIn // Use the latest value.
+          : overrideValueOut; // Override the value.
+      if (valueOutForUnitIn === undefined) {
+        // We don't have a quote, so abort.
+        return undefined;
+      }
+
+      /**
+       * @param {Amount} amountIn the given amountIn
+       * @returns {Amount} the amountOut that will be received
+       */
+      const calcAmountOut = amountIn => {
+        const valueIn = AmountMath.getValue(amountIn, brandIn);
+        return AmountMath.make(
+          floorDivide(multiply(valueIn, valueOutForUnitIn), unitIn),
+          brandOut,
+        );
+      };
+
+      /**
+       * @param {Amount} amountOut the wanted amountOut
+       * @returns {Amount} the amountIn needed to give
+       */
+      const calcAmountIn = amountOut => {
+        const valueOut = AmountMath.getValue(amountOut, brandOut);
+        return AmountMath.make(
+          ceilDivide(multiply(valueOut, unitIn), valueOutForUnitIn),
+          brandIn,
+        );
+      };
+
+      // Calculate the quote.
+      const quote = priceQuery(calcAmountOut, calcAmountIn);
+      if (!quote) {
+        return undefined;
+      }
+
+      const {
+        amountIn,
+        amountOut,
+        timestamp: theirTimestamp = timestamp,
+      } = quote;
+      AmountMath.coerce(amountIn, brandIn);
+      AmountMath.coerce(amountOut, brandOut);
+      if (theirTimestamp !== undefined) {
+        return authenticateQuote([
+          { amountIn, amountOut, timer, timestamp: theirTimestamp },
+        ]);
+      }
+      return E(timer)
+        .getCurrentTimestamp()
+        .then(now =>
+          authenticateQuote([{ amountIn, amountOut, timer, timestamp: now }]),
+        );
+    };
+
+  /**
+   * @param {bigint} _roundId
+   * @returns {boolean}
+   */
+  const timedOut = _roundId => {
+    if (!details.has(_roundId) || !rounds.has(_roundId)) {
+      return false;
+    }
+
+    const blockTimestamp = timer.getCurrentTimestamp();
```

Comment:
My advice to change `TimerService` to `ERef<TimerService>` will require you to use `E` and await this call.  You'll also need to change this function to async.

```suggestion
    const blockTimestamp = await E(timer).getCurrentTimestamp();
```